sdk/monitor/monitor-opentelemetry/src/traces/handler.ts

```diff
@@ -55,26 +54,22 @@ export class TraceHandler {
       exportTimeoutMillis: 30000,
       maxQueueSize: 2048,
     };
-    this._spanProcessor = new BatchSpanProcessor(this._azureExporter, bufferConfig);
+    this._batchSpanProcessor = new BatchSpanProcessor(this._azureExporter, bufferConfig);
+    this._azureSpanProcessor = new AzureMonitorSpanProcessor(this._metricHandler);
     this._azureFunctionsHook = new AzureFunctionsHook();
     this._initializeInstrumentations();
   }
 
-  public start(): void {
-    try {
-      const azureSpanProcessor = new AzureMonitorSpanProcessor(this._metricHandler);
-      (
-        (trace.getTracerProvider() as ProxyTracerProvider).getDelegate() as NodeTracerProvider
-      ).addSpanProcessor(azureSpanProcessor);
-    } catch (error) {}
-  }
-
   public getSampler(): ApplicationInsightsSampler {
     return this._aiSampler;
   }
 
-  public getSpanProcessor(): BatchSpanProcessor {
-    return this._spanProcessor;
+  public getBatchSpanProcessor(): BatchSpanProcessor {
+    return this._batchSpanProcessor;
+  }
+
+  public getAzureMonitorSpanProcessor(): AzureMonitorSpanProcessor {
+    return this._azureSpanProcessor;
```

Comment:
Why are there 2 processors now?

Reply:
There were always 2 processors, one take care of batching telemetry and sending to exporter and another take care of calculating data for standard and live metrics.